cardano-api/src/Cardano/Api/Orphans.hs

```diff
@@ -681,3 +778,1443 @@ instance Crypto.Crypto crypto => ToJSON (VMap VB VB (Shelley.KeyHash    'Shelley
+    , "endPeriod"   .= do kesEndPeriod    :: KESPeriod
+    , "evolution"   .= do kesEvolution    :: Word
+    ]
+
+instance ToJSON HotKey.KESEvolutionError where
+  toJSON (HotKey.KESCouldNotEvolve kesInfo targetPeriod) = object
+    [ "kind"          .= String "KESCouldNotEvolve"
+    , "kesInfo"       .= do kesInfo       :: HotKey.KESInfo
+    , "targetPeriod"  .= do targetPeriod  :: KESPeriod
+    ]
+  toJSON (HotKey.KESKeyAlreadyPoisoned kesInfo targetPeriod) = object
+    [ "kind"          .= String "KESKeyAlreadyPoisoned"
+    , "kesInfo"       .= do kesInfo       :: HotKey.KESInfo
+    , "targetPeriod"  .= do targetPeriod  :: KESPeriod
+    ]
+
+instance
+  ( ShelleyBasedEra era
+  , ToJSON (PredicateFailure (UTXO era))
+  , ToJSON (PredicateFailure (UTXOW era))
+  , ToJSON (PredicateFailure (Core.EraRule "BBODY" era))
+  , ToJSON (BlockTransitionError era)
+  ) => ToJSON (ShelleyLedgerError era) where
+  toJSON (BBodyError (BlockTransitionError fs)) = object
+    [ "kind"      .= String "BBodyError"
+    , "failures"  .= do map toJSON fs  :: [Value]
+    ]
+
+instance
+  ( ShelleyBasedEra era
+  , ToJSON (Ledger.PParamsDelta era)
+  , ToJSON (Crypto.OutputVRF (Core.VRF (Ledger.Crypto era)))
+  ) => ToJSON (ShelleyLedgerUpdate era) where
+  toJSON (ShelleyUpdatedProtocolUpdates updates) = object
+    [ "kind"    .= String "ShelleyUpdatedProtocolUpdates"
+    , "updates" .= do map toJSON updates :: [Value]
+    ]
+
+instance
+  ( Ledger.Era era, ToJSON (Ledger.PParamsDelta era)
+  , ToJSON (Crypto.OutputVRF (Core.VRF (Ledger.Crypto era)))
+  ) => ToJSON (ProtocolUpdate era) where
+  toJSON ProtocolUpdate{protocolUpdateProposal, protocolUpdateState} = object
+    [ "proposal" .= do protocolUpdateProposal :: UpdateProposal era
+    , "state"    .= do protocolUpdateState    :: UpdateState (Consensus.EraCrypto era)
+    ]
+
+instance ToJSON (Ledger.PParamsDelta era)
+         => ToJSON (UpdateProposal era) where
+  toJSON UpdateProposal{proposalParams, proposalVersion, proposalEpoch} = object
+    [ "params"  .= do proposalParams  :: Ledger.PParamsDelta era
+    , "version" .= do proposalVersion :: Maybe ProtVer
+    , "epoch"   .= do proposalEpoch   :: EpochNo
+    ]
+
+instance
+  ( Core.Crypto crypto
+  , ToJSON (Crypto.OutputVRF (Core.VRF crypto))
+  ) => ToJSON (UpdateState crypto) where
+  toJSON UpdateState{proposalVotes, proposalReachedQuorum} = object
+    [ "proposal"      .= do proposalVotes         :: [KeyHash 'Genesis crypto]
+    , "reachedQuorum" .= do proposalReachedQuorum :: Bool
+    ]
+
+instance
+  ( Core.Crypto crypto
+  , ToJSON (Crypto.CertifiedVRF (Core.VRF crypto) Nonce)
+  , ToJSON (Crypto.OutputVRF (Core.VRF crypto))
+  , ToJSON Ledger.ActiveSlotCoeff
+  ) => ToJSON (ChainTransitionError crypto) where
+  toJSON (ChainTransitionError fs) = object
+    [ "kind"      .= String "ChainTransitionError"
+    , "failures"  .= do map toJSON fs :: [Value]
+    ]
+
+instance ToJSON ChainPredicateFailure where
```

Comment:
So for example this `ToJSON` instance should not be necessary. Why? Because `ToObject ChainPredicateFailure` exists and the `ToObject` class has a `ToJSON` constraint: https://github.com/input-output-hk/iohk-monitoring-framework/blob/066f7002aac5a0efc20e49643fea45454f226caa/iohk-monitoring/src/Cardano/BM/Data/Tracer.lhs#L217
Can you double check that  the orphan instances you have added are needed?